File: asion-sample/asion-sample-core/src/main/java/org/asion/sample/SampleRepository.java
package org.asion.sample;

/**
 * @author Asion
 * @since 16/4/30.
 */
public interface SampleRepository {

	Iterable<Sample> findAll();

	Sample save(Sample sample);

	Sample findOne(Long id);

	void delete(Long id);

}

File: asion-sample/asion-sample-core/src/main/java/org/asion/sample/SampleRepository.kt
package org.asion.sample

/**
 * @author Asion
 * @since 16/4/30.
 */
interface SampleRepository {

    fun findAll(): Iterable<Sample>

    fun save(sample: Sample): Sample

    fun findOne(id: Long?): Sample

    fun delete(id: Long?)

}
